### ISSUE TITLE
"dyld: Library not loaded: /lib/libpython3.13.dylib" in 3.13.0a4 MacOS non-framework build with custom prefix

### ISSUE THREAD

**Issue by native-api** [python/cpython#115737]
# Bug report

### Bug description:

The above error was discovered in [Pyenv CI check](https://github.com/pyenv/pyenv/actions/runs/7927748950/job/21644695424?pr=2903) (logs are on the link).

See https://github.com/pyenv/pyenv/pull/2903#issuecomment-1954681384 for diagnostics and probable cause.

### CPython versions tested on:

3.13

### Operating systems tested on:

macOS

### Linked PRs
* gh-115750

**Comment by ronaldoussoren** [comment]
How can we reproduce this?  In particular, what is the command-line for configure and any additional customization of the build process?

**Comment by ned-deily** [comment]
From a quick look at the [linked pyenv issue](https://github.com/pyenv/pyenv/pull/2903#issuecomment-1954681384), it looks like this may be related to the changes for iOS frameworks builds (gh-114099). @freakboy3742?

**Comment by ronaldoussoren** [comment]
I'd still like to know how to reproduce this without having to hunt through the pyenv GitHub. The linked issue seems to use `-rpath` somehow, which is not something we explicitly support.

**Comment by freakboy3742** [comment]
@ned-deily As with @ronaldoussoren, I'm trying to work out what the reproduction instructions are... I'll dig in and see if I can reverse engineer the problem.

**Comment by native-api** [comment]
Attached a Bash trace log from the Pyenv invocation which contains configure/make calls and ennvar assignments. That's what I usually use to diagnose Pyenv issues.
[out.txt](https://github.com/python/cpython/files/14353562/out.txt)
If you need a more distilled view, I'll be able to compose it when I return from work in about 10 hours.

Setting rpaths is required in MacOS to find dylibs outside of the default search path -- in Homebrew and under an uncommon installation prefix.

**Comment by freakboy3742** [comment]
@native-api Thanks for that detail - based on that, I've been able work out the reproduction case is:
```
./configure --prefix=$(pwd)/root --enable-shared --libdir=$(pwd)/root/lib && make -j4 all install && $(pwd)/root/bin/python3.13 -m test.pythoninfo
```

**Comment by freakboy3742** [comment]
FWIW: it looks like the problem slipped in because while I did test non-framework builds, and non-framework builds in a non-default location, I didn't check a *shared* (i.e. ,``--enable-shared``) non-framework, non-default location build.

**Comment by ned-deily** [comment]
> _shared_ (i.e. ,`--enable-shared`) non-framework, non-default location build

Sorry, I should have thought to suggest testing that one as well originally. I generally avoid `--enable-shared` builds on macOS as they often lead to problems like this but they are used.

**Comment by native-api** [comment]
We have enabled `--enable-shared` by default by user requests. The rationale was it is required for products that link against `libpython` and it's the default in official CPython Docker images.

**Comment by ned-deily** [comment]
Thanks for the report and the followup info. The simplified fix is merged into main for release in 3.13.0a5.
